File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        result = []
        l, r = 0, len(numbers) - 1
        while (l < r):
            sumatory = numbers[l]+numbers[r]
            if (sumatory == target):
                result.append(l+1)
                result.append(r+1)
                return result
            if (sumatory <= target):
                l += 1
            if (sumatory >= target):
                r -= 1

        return result 

File: Data Structures & Algorithms/two-integer-sum/submission-0.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] result = new int[2];
        Map<Integer, Integer> storeIndex = new HashMap<>();

        for (int i = 0 ; i < nums.length; i++) {
            if (storeIndex.containsKey(target - nums[i])) {
                result[0] = storeIndex.get(target - nums[i]);
                result[1] = i;
                return result;
            }
            storeIndex.put(nums[i], i);
        }



        return result;
    }
}
